articles/scale_on_aws.rst

```diff
@@ -0,0 +1,288 @@
+engineering resources. (Honestly, if the founding engineers of the
+Heroku PostgreSQL team didn’t join Citus, we might have decided to wait
+on this.) Since making Citus Cloud publicly available, we are now more
+bullish on the cloud then ever.
+
+It turns out that targeting an important use case for your customers and
+delivering it to them in a way that removes their pain points, matters
+more than anything else. In this article, we’ll only focus on removing
+operational pain points and not on use cases: Why is cloud changing the
+way databases are delivered to customers? What AWS technologies Citus
+Cloud is using to enable that in a unique way?
+
+The following sections highlight our learnings from talking to hundreds
+of prospective customers. Early on, we thought that our customers were
+asking for something that was more magical than attainable -- Citus
+should handle failovers in a way that’s transparent to the application,
+elastically scale up and scale out, take automatic backups for disaster
+recovery, have monitoring and alarms built in, and provide patches and
+upgrades with no downtime.
+
+We then started breaking down each customer conversation into smaller
+pieces and found that our customers had seven operational pain points.
+In the following, we’re going to talk about each one of these seven
+points as a mini-learning. Although we describe these learnings in the
+context of PostgreSQL, Citus, and AWS, they are also applicable to other
+databases and cloud vendors.
+
+Since Citus is an extension to PostgreSQL (not a fork), it makes sense
+to start with a pain most PostgreSQL users have.
+
+1. High Availability
+~~~~~~~~~~~~~~~~~~~~
+
+PostgreSQL provides `streaming
+replication <https://www.postgresql.org/docs/current/static/warm-standby.html>`__
+as a way to provide high availability since its v9.0 release. Streaming
+replication works by designating one PostgreSQL instance as the primary
+node. The primary node takes all writes and executes all changes against
+the database. It then ships those changes to one or more secondary nodes
+for them to apply.
+
+When you talk to users who use streaming replication in production, they
+express two pain points with it. The first is the complexity associated
+with it: `it takes twelve
+steps <https://wiki.postgresql.org/wiki/Streaming_Replication>`__ to
+setup streaming replication, and you may also need to understand
+concepts such as the number of wal segments and replication lag in the
+process.
+
+The second and more important one relates to auto failover. If the
+primary node fails, how do you detect the failure and promote a
+secondary to be the new primary? (In distributed systems, this is also
+known as the leader election problem.) To enable this, you need a
+monitoring component that watches the primary and secondary nodes and
+reliably initiates their failovers.
+
+.. image:: ../images/citus-cloud-diagrams.png
```

Comment:
Small question. Is there a way to "center" the images in this article?

Reply:
(Done)